### PR TITLE
pidref: add pidref_parent() helper and us it in userdbd

### PR DESCRIPTION
**Pull request by poettering**
let's continue our pidfd'ification, and add a race-free pidref wrapper around getppid()

### CODE REVIEW

File: src/basic/pidref.c [hunk 1]
```diff
@@ -106,6 +106,38 @@ int pidref_set_pidfd_consume(PidRef *pidref, int fd) {
         return r;
 }
 
+int pidref_set_parent(PidRef *pidref) {
```
Comment (yuwata) [inline src/basic/pidref.c]: Why not to name `ret` ?

**Review by yuwata — APPROVED** [review]
LGTM. One question.

File: src/userdb/userwork.c [hunk 2]
```diff
@@ -554,14 +561,11 @@ static int run(int argc, char *argv[]) {
...
-                                if (parent <= 1)
-                                        return log_error_errno(SYNTHETIC_ERRNO(ESRCH), "Parent already died?");
-
-                                if (kill(parent, SIGUSR2) < 0)
-                                        return log_error_errno(errno, "Failed to kill our own parent: %m");
+                                r = pidref_kill(&parent, SIGUSR2);
+                                if (r == -ESRCH)
+                                        return log_error_errno(r, "Parent already died?");
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to send SIGUSR2 signal to parent. %m");
```
Comment (YHNdnzj) [inline src/userdb/userwork.c]:
```suggestion
                                        return log_error_errno(r, "Failed to send SIGUSR2 signal to parent: %m");
```

**Review by YHNdnzj — APPROVED** [review]

File: src/basic/pidref.h [hunk 3]
```diff
@@ -38,7 +38,7 @@ int pidref_set_pidstr(PidRef *pidref, const char *pid);
 int pidref_set_pidfd(PidRef *pidref, int fd);
 int pidref_set_pidfd_take(PidRef *pidref, int fd); /* takes ownership of the passed pidfd on success*/
 int pidref_set_pidfd_consume(PidRef *pidref, int fd); /* takes ownership of the passed pidfd in both success and failure */
-
+int pidref_set_parent(PidRef *pidref);
```
Comment (yuwata) [inline src/basic/pidref.h]: Then, `pidref` -> `ret`

**Review by yuwata — APPROVED** [review]

**Comment by poettering** [comment]
made the suggested change. will merge right-away because so trivial and CI so slow and overloaded.

**Pull request merged**